### PR TITLE
Switch from rcdom to lol_html

### LINKED ISSUE #79: Upgrade to HTML5ever 0.25
This is not trivial because of https://github.com/servo/html5ever/issues/407. `lol_html` worked pretty well for docs.rs, maybe that would work here too?

### PR DESCRIPTION
**Pull request by jyn514**
This is both more efficient and easier to read. Unfortunately it
introduces a ton of dependencies.

Closes #79

### CODE REVIEW

**Comment by jyn514** [comment]
Ok, I think this is ready for review.

r? @zregvart

**Comment by jyn514** [comment]
Huh, surprisingly this didn't affect build times almost at all, even though there are 15 more dependencies now.

Before: https://jyn514.github.io/assets/cargo-timings-html5ever.html
After: https://jyn514.github.io/assets/cargo-timing-lol_html.html

File: src/parse.rs [hunk 1]
```diff
@@ -53,36 +41,21 @@ fn parse_a_hrefs(handle: &Handle, root_url: &Url, base_url: &Url, urls: &mut Has
...
-    }
+    urls
 }
 
 #[cfg(test)]
 mod test {
-    use html5ever::parse_document;
-    use html5ever::{rcdom::RcDom, tendril::TendrilSink};
-    use std::collections::HashSet;
+    use super::parse_a_hrefs as gather_urls;
```
Comment (zregvart) [inline src/parse.rs:56]: Nitpick: perhaps use it as `parse_a_hrefs` and refactor `gather_urls` to `parse_a_hrefs` below?

**Review by zregvart — APPROVED** [review]
Thanks, this is indeed cleaner. LGTM, one not at all important nitpick.

**Pull request merged**